### PR TITLE
Import lockbox-ios-fxa-sync via carthage

### LINKED ISSUE #241: Build lockbox-ios with fxa-sync and storage
Follow up to #141

Smash together the two code bases, "minimally".

- Get it compiling
- Included/linked in the app

### PR DESCRIPTION
**Pull request by jhugman**
Resolves #241

This PR imports lockbox-ios-fxa-sync from a branch suited for importing.

It also trivially uses a `Profile`, and `FxALoginHelper` in the `AppDelegate`.

This initializes a sqlcipher database for Logins. No UI is affected.

Once the [lockbox-ios-fxa-sync branch][1] is merged, an additional change to `Carthage.resolved` can be made before this PR should be landed.

[1]: https://github.com/mozilla-lockbox/lockbox-ios-fxa-sync/pull/2

### CODE REVIEW

File: Cartfile.resolved [hunk 1]
```diff
@@ -3,4 +3,12 @@ github "Quick/Quick" "v1.2.0"
 github "ReactiveX/RxSwift" "4.1.2"
 github "RxSwiftCommunity/RxDataSources" "3.0.2"
 github "RxSwiftCommunity/RxOptional" "3.4.1"
+
```
Comment (sashei) [inline Cartfile.resolved]: 👾

File: Cartfile.resolved [hunk 2]
```diff
@@ -3,4 +3,12 @@ github "Quick/Quick" "v1.2.0"
...
 github "mozilla-mobile/telemetry-ios" "v1.0.4"
+
+github "mozilla-lockbox/lockbox-ios-fxa-sync" "e35a65f23aa6be7f435a16c32703d05bd277a8de"
+github "sleroux/Deferred" "35b8927c1b94ce074e10793c57e1f80d0e2227fa"
+github "jrendel/SwiftKeychainWrapper" "3.0.1"
+github "SwiftyJSON/SwiftyJSON" "3.1.4"
```
Comment (sashei) [inline Cartfile.resolved]: 😢

**Comment by devinreams** [comment]
@jhugman in case you didn't spot it yet: due to recent merges on `master` your branch has a conflict in the `project.pbxproj` file now and needs to be resolved before buddybuild tests can re-build and test. ❌

**Comment by jhugman** [comment]
Rebased to master, and addressed review comments.

**Comment by jhugman** [comment]
Tests are timing out at `WebViewSpec.swift`.

**Pull request merged**